pkg/curatedpackages/kubeclient_test.go

```diff
@@ -0,0 +1,24 @@
+package curatedpackages
```

Comment:
So we don't want to make this a test package, like `curatedpackages_test`? I understand that we are making using of `restConfigurator`, but if this abstraction improves testing, we can still have that comment and make it exported. Just curious if there is any other reason we aren't exporting that function. 

I see that we are calling `newKubeClient` as well. Not sure what you and @g-gaston discussed about this, but this can be a situation where we set the restConfigurator after calling the initial constructor.

Reply:
I don't export restConfigurator because no one else needs it. Each name that's exported is something that could break someone else, often in surprising ways. I try not to export something without a good reason, and even then, something this small, and laser focused—I'd be just as likely to copy it as I would be to export it. I find its preferable to reduce my dependency surface vs remove 5 lines of duplicated code. There's also something to be said for having the definition close at hand, as opposed to in a file X packages away.

As for newKubeClient, I think you were included in the Slack thread, so you can see what we discussed. Yes, we could set restConfigurator after the fact, but felt that this approach melds the best of both worlds. The only place that has any reason to set a restConfigurator is a test, and so we elected to keep any mention of restConfigurator or anything to manipulate it, out of the exported interface. Normal users should not have any reason to modify the restConfigurator, and they shouldn't even have to know it exists.